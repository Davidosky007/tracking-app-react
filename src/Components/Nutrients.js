import PropTypes from 'prop-types';
import { icons, measurementUnits } from '../logic/info';

const Nutrient = (props) => {
  const {
    unit, measurements, getMeasurementsByDate, reduceMethod, selectedDate,
  } = props;
  const measurementsByDate = getMeasurementsByDate(measurements, selectedDate);
  const displayedValue = reduceMethod(measurementsByDate);

  return (
    <div className="flex nutrient center p-l-20 p-r-20 m-b-50">
      <div className={`${icons[unit]} n-icon x-big color-blue text-center`} />
      <div className="n-details m-l-20">
        <div className="color-gray overflow-hidden">{unit}</div>
        <div className="color-gray m-big">
          {displayedValue}
          <span className="x-small">{`(${measurementUnits[unit]})`}</span>
        </div>
      </div>
    </div>
  );
};

Nutrient.propTypes = {
  unit: PropTypes.string.isRequired,
  measurements: PropTypes.arrayOf(PropTypes.object).isRequired,
  getMeasurementsByDate: PropTypes.func.isRequired,
  reduceMethod: PropTypes.func.isRequired,
  selectedDate: PropTypes.shape({}).isRequired,
};

export default Nutrient;
