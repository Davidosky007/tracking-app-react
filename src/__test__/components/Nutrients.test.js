import renderer from 'react-test-renderer';
import Nutrients from '../../Components/Nutrients';

describe('Nutrients', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Nutrients
      unit=""
      measurements={[{}]}
      getMeasurementsByDate={() => {}}
      reduceMethod={() => {}}
      selectedDate={{}}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
