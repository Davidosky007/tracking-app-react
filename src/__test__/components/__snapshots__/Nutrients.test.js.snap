// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nutrients renders correctly 1`] = `
<div
  className="flex nutrient center p-l-20 p-r-20 m-b-50"
>
  <div
    className="undefined n-icon x-big color-blue text-center"
  />
  <div
    className="n-details m-l-20"
  >
    <div
      className="color-gray overflow-hidden"
    >
      
    </div>
    <div
      className="color-gray m-big"
    >
      <span
        className="x-small"
      >
        (undefined)
      </span>
    </div>
  </div>
</div>
`;
